python/cudf/cudf/__init__.py

```diff
@@ -1,9 +1,63 @@
 # Copyright (c) 2018-2019, NVIDIA CORPORATION.
+""" __init__.py
 
+   isort:skip_file
+"""
 import cupy
 
-import rmm
+from cudf.errors import UnSupportedGPUError, UnSupportedCUDAError
+
+gpus_count = cupy.cuda.runtime.getDeviceCount()
+
+
+if gpus_count > 0:
+    for device in range(0, gpus_count):
+        # cudaDevAttrComputeCapabilityMajor - 75
+        major_version = cupy.cuda.runtime.deviceGetAttribute(75, device)
+
+        if major_version >= 6:
+            # You have a GPU with NVIDIA Pascal™ architecture or better
+            pass
+        else:
+            raise UnSupportedGPUError(
+                "You will need a GPU with NVIDIA Pascal™ architecture or better"
+            )
+
+    cuda_runtime_version = cupy.cuda.runtime.runtimeGetVersion()
 
+    if cuda_runtime_version > 10000:
+        # CUDA Runtime Version Check: Runtime version is greater than 10000
+        pass
+    else:
+        raise UnSupportedCUDAError(
+            "Please update your CUDA Runtime to 10.0 or above"
+        )
+
+    cuda_driver_version = cupy.cuda.runtime.driverGetVersion()
+
+    if cuda_driver_version == 0:
+        raise UnSupportedCUDAError("Please install CUDA Driver")
+    elif cuda_driver_version >= cuda_runtime_version:
+        # CUDA Driver Version Check: Driver Runtime version is >= Runtime version
+        pass
+    else:
+        raise UnSupportedCUDAError(
+            "The detected driver version does not support the detected CUDA Runtime version. Please update your NVIDIA GPU Driver.\n"
+            "Detected CUDA Runtime version : "
+            + str(cuda_runtime_version)
+            + "\n"
+            "Latest version of CUDA supported by current NVIDIA GPU Driver : "
+            + str(cuda_driver_version)
+        )
+
+else:
+    import warnings
+
+    warnings.warn(
+        "You donot have an NVIDIA GPU, please install one and try again"
+    )
```

Comment:
Move all of this into a utility function somewhere and call that utility function to clean up this file.

Reply:
Cleaned it up 👍